routing/notifications.go

```diff
@@ -113,15 +113,19 @@ func (r *ChannelRouter) notifyTopologyChange(topologyDiff *TopologyChange) {
 	r.RLock()
 	numClients := len(r.topologyClients)
 	r.RUnlock()
-	if numClients != 0 {
-		log.Tracef("Sending topology notification to %v clients %v",
-			numClients,
-			newLogClosure(func() string {
-				return spew.Sdump(topologyDiff)
-			}),
-		)
+
+	// Do not reacquire the lock twice unnecessarily.
```

Comment:
Since we're already here, we could simplify this further to the following:
```
r.RLock()
defer r.RUnlock()

if len(r.topologyClients) == 0 {
    return
}

log.Tracef(...)

for _, client := range r.topologyClients {
    ...
}
```

Reply:
@wpaulino 

It is simpler but less granular. My goal was to limit the time spent under the lock.

Reply:
Your changes still reacquire the lock in the case that we have topology clients to notify.

Reply:
Second @wpaulino's suggestion 👍